### PR TITLE
IGNITE-23283 Throw CompactedException for single get metastorage methods

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-23283

### CODE REVIEW

File: modules/metastorage/src/testFixtures/java/org/apache/ignite/internal/metastorage/server/SimpleInMemoryKeyValueStorage.java [hunk 1]
```diff
@@ -740,22 +741,26 @@ private Collection<Entry> doGetAll(List<byte[]> keys, long rev) {
     }
 
     private Entry doGet(byte[] key, long revUpperBound) {
-        assert revUpperBound >= 0 : "Invalid arguments: [revUpperBound=" + revUpperBound + ']';
```
Comment (rpuch) [inline modules/metastorage/src/testFixtures/java/org/apache/ignite/internal/metastorage/server/SimpleInMemoryKeyValueStorage.java:743]: Let's revert this change (information about `revUpperBound` seems to be useful)

File: modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/exceptions/CompactedException.java [hunk 2]
```diff
@@ -36,10 +36,17 @@ public CompactedException() {
     /**
      * Constructs an exception with a given message.
      *
-     * @param revision Requested revision.
+     * @param requestedRevision Requested revision.
+     * @param latestCompatedRevision Latest compacted revision.
      */
-    public CompactedException(long revision) {
-        super(COMPACTED_ERR, "Requested revision has already been compacted: " + revision);
+    public CompactedException(long requestedRevision, long latestCompatedRevision) {
```
[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/exceptions/CompactedException.java:42]: Thanks

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java [hunk 3]
```diff
@@ -1076,35 +1078,27 @@ private Collection<Entry> doGetAll(Collection<byte[]> keys, long rev) {
...
 
-    /**
-     * Gets the value by key and revision.
-     *
-     * @param key            Target key.
-     * @param revUpperBound  Target upper bound of revision.
-     * @return Value.
-     */
     private Entry doGet(byte[] key, long revUpperBound) {
-        assert revUpperBound >= 0 : "Invalid arguments: [revUpperBound=" + revUpperBound + ']';
```
[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java:1087]: I tried to fix it differently.

File: modules/metastorage/src/testFixtures/java/org/apache/ignite/internal/metastorage/server/SimpleInMemoryKeyValueStorage.java (hunk 1 above)
Comment (tkalkirill) [reply modules/metastorage/src/testFixtures/java/org/apache/ignite/internal/metastorage/server/SimpleInMemoryKeyValueStorage.java:728]: I tried to fix it differently.

**Review by rpuch — APPROVED** [review]
